### ISSUE TITLE
Unable to parse newer .wast

### ISSUE THREAD

**Issue by juj** [bathos/wast-sublime-syntax#1]
Hi, thanks for developing this plugin!

Tried it out today, and observed that the following .wast gave some trouble:

```wast
(module
 (type $none_=>_none (func))
 (type $none_=>_i32 (func (result i32)))
 (type $i32_i32_=>_i32 (func (param i32 i32) (result i32)))
 (type $i32_i32_i32_=>_i32 (func (param i32 i32 i32) (result i32)))
 (type $none_=>_f64 (func (result f64)))
 (import "a" "a" (memory $2 256 32768))
 (data (i32.const 1024) "return parseInt(location.search.substr(1) || \'12346\')\00\00document.body.innerHTML += \'Num T shapes in a lattice grid of size \' + $0 + \': \' + $1 + \'. (calculated in \' + $2.toFixed(3) + \' msecs)<br>\'\00idd")
 (import "a" "b" (func $emscripten_performance_now (result f64)))
 (import "a" "c" (func $emscripten_asm_const_int (param i32 i32 i32) (result i32)))
 (table $0 1 1 funcref)
 (global $global$0 (mut i32) (i32.const 5244112))
 (export "d" (table $0))
 (export "e" (func $__wasm_call_ctors))
 (export "f" (func $main))
 (func $main (; 2 ;) (param $0 i32) (param $1 i32) (result i32)
  (call $__original_main)
 )
 (func $__wasm_call_ctors (; 3 ;)
  (nop)
 )
 (func $__original_main (; 4 ;) (result i32)
  (local $0 i32)
  (local $1 i32)
  (local $2 i32)
  (local $3 i32)
  (local $4 i32)
  (local $5 i32)
  (local $6 i32)
  (local $7 i32)
  (local $8 i32)
  (local $9 i32)
  (local $10 i32)
  (local $11 i32)
  (local $12 i64)
  (local $13 i64)
  (local $14 i64)
  (local $15 i64)
  (local $16 f64)
  (global.set $global$0
   (local.tee $6
    (i32.sub
     (global.get $global$0)
     (i32.const 32)
    )
   )
  )
  (i32.store
   (i32.const 1224)
   (call $emscripten_asm_const_int
    (i32.const 1024)
    (i32.const 1078)
    (i32.const 0)
   )
  )
  (local.set $16
   (call $emscripten_performance_now)
  )
  (block $label$1
   (br_if $label$1
    (i32.lt_s
     (local.tee $3
      (i32.load
       (i32.const 1224)
      )
     )
     (i32.const 4)
    )
   )
   (local.set $0
    (i32.add
     (local.get $3)
     (i32.const -3)
    )
   )
   (local.set $7
    (i32.add
     (i32.div_u
      (i32.add
       (local.get $3)
       (i32.const -4)
      )
      (i32.const 3)
     )
     (i32.const 1)
    )
   )
   (local.set $1
    (i32.const 1)
   )
   (loop $label$2
    (local.set $13
     (i64.add
      (local.get $13)
      (i64.extend_i32_u
       (i32.mul
        (local.get $0)
        (local.get $0)
       )
      )
     )
    )
    (local.set $5
     (i32.ne
      (local.get $1)
      (local.get $7)
     )
    )
    (local.set $0
     (i32.add
      (i32.mul
       (local.tee $1
        (i32.add
         (local.get $1)
         (i32.const 1)
        )
       )
       (i32.const -3)
      )
      (local.get $3)
     )
    )
    (br_if $label$2
     (local.get $5)
    )
   )
   (br_if $label$1
    (i32.lt_s
     (local.get $3)
     (i32.const 6)
    )
   )
   (local.set $0
    (i32.add
     (local.get $3)
     (i32.const -5)
    )
   )
   (local.set $7
    (i32.add
     (i32.div_u
      (i32.add
       (local.get $3)
       (i32.const -6)
      )
      (i32.const 5)
     )
     (i32.const 1)
    )
   )
   (local.set $1
    (i32.const 1)
   )
   (loop $label$3
    (local.set $12
     (i64.add
      (local.get $12)
      (i64.extend_i32_u
       (i32.mul
        (local.get $0)
        (local.get $0)
       )
      )
     )
    )
    (local.set $5
     (i32.ne
      (local.get $1)
      (local.get $7)
     )
    )
    (local.set $0
     (i32.add
      (i32.mul
       (local.tee $1
        (i32.add
         (local.get $1)
         (i32.const 1)
        )
       )
       (i32.const -5)
      )
      (local.get $3)
     )
    )
    (br_if $label$3
     (local.get $5)
    )
   )
  )
  (local.set $12
   (i64.shl
    (i64.add
     (local.get $12)
     (local.get $13)
    )
    (i64.const 2)
   )
  )
  (local.set $1
   (i32.const 1)
  )
  (loop $label$4
   (local.set $10
    (i32.shl
     (local.get $1)
     (i32.const 1)
    )
   )
   (local.set $5
    (i32.mul
     (local.get $1)
     (i32.const 3)
    )
   )
   (local.set $0
    (local.tee $7
     (i32.add
      (local.get $1)
      (i32.const 1)
     )
    )
   )
   (local.set $13
    (local.tee $14
     (local.get $12)
    )
   )
   (loop $label$5
    (local.set $13
     (i64.add
      (local.tee $12
       (local.get $13)
      )
      (i64.shl
       (local.tee $15
        (i64.mul
         (i64.extend_i32_u
          (select
           (local.tee $4
            (i32.sub
             (local.get $3)
             (select
              (local.tee $2
               (i32.add
                (local.get $0)
                (local.get $5)
               )
              )
              (local.tee $4
               (select
                (local.get $5)
                (local.tee $4
                 (select
                  (local.tee $4
                   (i32.add
                    (local.tee $8
                     (i32.mul
                      (local.get $0)
                      (i32.const 3)
                     )
                    )
                    (local.get $10)
                   )
                  )
                  (local.tee $11
                   (select
                    (local.tee $9
                     (i32.add
                      (local.get $1)
                      (local.get $8)
                     )
                    )
                    (local.get $0)
                    (i32.lt_u
                     (local.get $0)
                     (local.get $9)
                    )
                   )
                  )
                  (i32.gt_s
                   (local.get $4)
                   (local.get $11)
                  )
                 )
                )
                (i32.lt_s
                 (local.get $4)
                 (local.get $5)
                )
               )
              )
              (i32.gt_s
               (local.get $2)
               (local.get $4)
              )
             )
            )
           )
           (i32.const 0)
           (i32.gt_s
            (local.get $4)
            (i32.const 0)
           )
          )
         )
         (i64.extend_i32_u
          (select
           (local.tee $2
            (i32.sub
             (local.get $3)
             (select
              (local.get $9)
              (local.tee $2
               (select
                (local.get $8)
                (local.tee $2
                 (select
                  (local.tee $4
                   (i32.add
                    (i32.shl
                     (local.get $0)
                     (i32.const 1)
                    )
                    (local.get $5)
                   )
                  )
                  (local.tee $2
                   (select
                    (local.get $2)
                    (local.get $1)
                    (i32.lt_u
                     (local.get $1)
                     (local.get $2)
                    )
                   )
                  )
                  (i32.lt_s
                   (local.get $2)
                   (local.get $4)
                  )
                 )
                )
                (i32.lt_s
                 (local.get $2)
                 (local.get $8)
                )
               )
              )
              (i32.lt_s
               (local.get $2)
               (local.get $9)
              )
             )
            )
           )
           (i32.const 0)
           (i32.gt_s
            (local.get $2)
            (i32.const 0)
           )
          )
         )
        )
       )
       (i64.const 3)
      )
     )
    )
    (local.set $0
     (i32.add
      (local.get $0)
      (i32.const 1)
     )
    )
    (br_if $label$5
     (i64.ne
      (local.get $15)
      (i64.const 0)
     )
    )
   )
   (local.set $1
    (local.get $7)
   )
   (br_if $label$4
    (i64.ne
     (local.get $12)
     (local.get $14)
    )
   )
  )
  (f64.store offset=16
   (local.get $6)
   (f64.sub
    (call $emscripten_performance_now)
    (local.get $16)
   )
  )
  (i32.store
   (local.get $6)
   (i32.load
    (i32.const 1224)
   )
  )
  (f64.store offset=8
   (local.get $6)
   (f64.convert_i64_u
    (local.get $14)
   )
  )
  (drop
   (call $emscripten_asm_const_int
    (i32.const 1079)
    (i32.const 1219)
    (local.get $6)
   )
  )
  (global.set $global$0
   (i32.add
    (local.get $6)
    (i32.const 32)
   )
  )
  (i32.const 0)
 )
)
```

First on the line ` (table $0 1 1 funcref)`, but then deleting that from the text file, it does not like the first i32.store it sees:
```
  (i32.store
   (i32.const 1224)
   (call $emscripten_asm_const_int
    (i32.const 1024)
    (i32.const 1078)
    (i32.const 0)
   )
  )
```

Very useful plugin nevertheless! Nice work :)

**Comment by bathos** [comment]
Thanks for the sample. It’s been a while since I looked at this so it might take me some time to get my head back in it. You said this is “newer” wast — do you know specifically what additions might have happened since this was last updated? (Though this might have always been a bug; I don’t recall if this was or wasn’t valid syntax at the time and it hasn’t been tested extensively.)

**Comment by juj** [comment]
Hmm, when writing this up, I thought `(table $0 1 1 funcref)` would have been of new syntax (https://github.com/WebAssembly/reference-types/blob/master/proposals/reference-types/Overview.md will add `funcref` support as a value type), but I was confused, since `funcref`s as part of `table` have always been part of the MVP and reference types are not used here.

**Comment by whitequark** [comment]
This is fixed as of [release 1.3.0](https://github.com/bathos/wast-sublime-syntax/releases/tag/1.3.0).
